pkg/component/observability/monitoring/persesoperator/deployment.go

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package persesoperator
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+const deploymentName = "perses-operator"
+
+func (p *persesOperator) deployment() *appsv1.Deployment {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: p.namespace,
+			Labels:    GetLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             ptr.To[int32](1),
+			RevisionHistoryLimit: ptr.To[int32](2),
+			Selector:             &metav1.LabelSelector{MatchLabels: GetLabels()},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: utils.MergeStringMaps(GetLabels(), map[string]string{
+						v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+					}),
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName:            serviceAccountName,
+					PriorityClassName:             p.values.PriorityClassName,
+					TerminationGracePeriodSeconds: ptr.To[int64](10),
```

Comment:
What is the reasoning for specifying `TerminationGracePeriodSeconds`?

Reply:
Inherited from the upstream [repo](https://github.com/perses/perses-operator/blame/6855a844c9e680c5cab3772e88cdac110e05bf2c/config/manager/manager.yaml#L70). Seeing how the commit that added it is the initial SDK generation, I would hazard a guess that it's fine to remove.

Reply:
We use `TerminationGracePeriodSeconds` for other components. Hence, it can also stay, but we should be aware why it is there as it may cause some trouble by delaying operations. I thought the operator is mainly stateless and hence it would not be needed, but I may be wrong.